src/components/TimeGrid.jsx

```diff
@@ -0,0 +1,238 @@
+import React, { PropTypes, Component } from 'react'
+import cn from 'classnames'
+import localizer from '../localizer.js'
+import formats from '../formats.js'
+import { findDOMNode } from 'react-dom';
+import classes from 'dom-helpers/class';
+import scrollbarSize from 'dom-helpers/util/scrollbarSize';
+import getWidth from 'dom-helpers/query/width';
+
+import TimeGridHeader from './TimeGridHeader.jsx'
+import TimeGutter from './TimeGutter.jsx'
+import DaySlot from './DaySlot.jsx'
+import TimeGridAllDay from './TimeGridAllDay.jsx'
+
+import EventRow from '../EventRow.jsx'
+import EventCell from '../EventCell.jsx'
+
+import { accessor as get } from '../utils/accessors';
+import dates from '../utils/dates';
+import {
+  inRange, eventSegments, endOfRange
+  , eventLevels, sortEvents, segStyle } from '../utils/eventLevels';
+import { elementType, accessor } from '../utils/propTypes';
+
+const MIN_ROWS = 2;
+
+export default class TimeGrid extends Component {
+  static propTypes = {
+    ...DaySlot.propTypes,
+    ...TimeGutter.propTypes,
+
+    end: PropTypes.instanceOf(Date),
+    start: PropTypes.instanceOf(Date),
+    selectRangeFormat: PropTypes.func.isRequired,
+    eventTimeRangeFormat: PropTypes.func.isRequired,
+    timeGutterFormat: PropTypes.string,
+    dayFormat: PropTypes.string.isRequired,
+    culture: PropTypes.string.isRequired,
+    components: PropTypes.shape({
```

Comment:
they don't use Agenda, month, or toolbar components, so they should be specified here. At this point each component should be passed in individually as `eventComponent`, and so on. In any case any props from EventRow should already be covered in the spread of DaySlot and TimeGutter above